lib/req-res/responder.js

```diff
@@ -38,43 +39,63 @@ util.inherits( Responder, Rabbus.Responder );
 
 // Instance Methods
 // ----------------
-Responder.prototype.inbound = function ( message, respond, callback ) {
-	const statusOptions = {
-		respond,
-		message,
-		'log'         : this.logger,
-		'messageType' : this.messageType,
-		'emitter'     : this
-	};
+Responder.prototype.inbound = function ( message, send, statusOptions, callback ) {
+	return new Promise( function ( resolve, reject ) {
+		this.on( 'error', function ( consumeError ) {
+			reject( consumeError );
+		} );
 
-	if ( !this.vSchema || this.vSchema === 'undefined' ) {
-		// if no validation, invoke callback immediately
-		callback( message, requestStatus( statusOptions ) );
-	} else {
-		// perform validation if there is a validation schema
-		validate( {
-			'value'   : message,
-			'schema'  : this.vSchema,
-			'options' : this.vOptions
-		} )
-		.then( function ( data ) {
-			callback( data, requestStatus( statusOptions ) );
-		} )
-		.catch( function ( error ) {
-			// send fail
-			respond( {
-				'status' : 'fail',
-				'data'   : error.message
+		if ( !this.vSchema || this.vSchema === 'undefined' ) {
+			// if no validation, invoke callback immediately
+			callback( message, send );
+		} else {
+			// perform validation if there is a validation schema
+			validate( {
+				'value'   : message,
+				'schema'  : this.vSchema,
+				'options' : this.vOptions
+			} )
+			.then( function ( data ) {
+				callback( data, send );
+			} )
+			.catch( function ( error ) {
+				// check if it's validation error
+				if ( error.validate ) {
+					send.fail( error.message );
+					return clearTimeout( statusOptions.timeout );
+				}
+				reject( error );
 			} );
-		} );
-	}
+		}
+	}.bind( this ) );
 };
 
 Responder.prototype.consume = function ( callback ) {
 	const that = this;
 
 	this.handle( function ( message, respond ) {
-		that.inbound( message, respond, callback );
+		const statusOptions = {
+			'log'         : that.logger,
+			'messageType' : that.messageType,
+			respond,
+			message,
+			'emitter'     : that,
+			'timeout'     : config.timeout.set( {
+				respond,
+				message,
+				'log'         : that.logger,
+				'emitter'     : that,
+				'messageType' : that.messageType
+			} )
+		};
+
+		const send = requestStatus( statusOptions );
+
+		that.inbound( message, send, statusOptions, callback )
+		.catch( ( consumeError ) => {
+			send.error( consumeError.message, consumeError );
```

Comment:
Does that matter? She's returning the very next [line](https://github.com/yamii/lapin/blob/511cc2fb4e7eababc7e573f05325c2573d765a94/lib/req-res/responder.js#L65), which mean's it not going to call reject on [line 65](https://github.com/yamii/lapin/blob/511cc2fb4e7eababc7e573f05325c2573d765a94/lib/req-res/responder.js#L67). Doesn't mean we won't have a rejected promise, and this .catch won't get executed?